### ISSUE TITLE
cannot find kernel in interactive window

### ISSUE THREAD

**Issue by zesluo** [microsoft/vscode-jupyter#6316]
## Environment data

Version: 1.57.0 (user setup)
Commit: b4c1bd0a9b03c749ea011b06c6d2676c8091a70c
Date: 2021-06-09T17:18:42.354Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19043

## Expected behaviour

XXX

## Actual behaviour

XXX

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

I just tried the latest version of jupyter and ms-python extension. It couldn't show the kernels as pic2. 'pysparkkernel', 'synape_pyspark'

![image](https://user-images.githubusercontent.com/46309533/122403660-1dbcb700-cfb1-11eb-93ce-cc5b31e6b188.png)
![image](https://user-images.githubusercontent.com/46309533/122403805-3af18580-cfb1-11eb-9aa9-8f3ca3e6821e.png)
Also, I tried to do some workaround, such as sticking to an older version of jupyter, python. Previously, it worked. But now it pops out a notification like this:

![image](https://user-images.githubusercontent.com/46309533/122403872-4c3a9200-cfb1-11eb-919d-fcf056a3498b.png)
![image](https://user-images.githubusercontent.com/46309533/122403932-5a88ae00-cfb1-11eb-95f6-b313d7860dc4.png)

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by DonJayamanne** [comment]
Thanks for filling this issue.
The reason things aren't working is because you need to update your extensions. Looking at the screenshot, you need to update your version of Jupyter extension (you have some pending updates).
Please update the extensions, I'm sure that'll fix the issues.
Please feel free to reopen or create a new issue if that doesn't work

**Comment by zesluo** [comment]
Hello @DonJayamanne ,  Thanks for your reply! @rchiodo  1.The first two pics show that both of jupyter and ms-python extensions are in the latest version. And this cannot work since it couldn't show the kernel : pysparkkernel', 'synape_pyspark'. 2. The last two pics show that  it couldn't work as well even in an older version of jupyter and ms-python extensions. Please kindly reopen the issue. we get emails from customers who are blocked by this. Thanks in advance!

**Comment by DonJayamanne** [comment]
Please could you send the following information;
* Version of python
* Is conda or  virtual env or other
* logs from the Jupyter output panel
* Screenshot of the errors

**Comment by rchiodo** [comment]
If I'm remembering right, I believe the problem here was that there are two installs of jupyter and we're only finding the kernelspecs for one and not the other? 

@zesluo can you describe the repro steps? Do we just have to install the Synapse extension?

**Comment by zesluo** [comment]
> Please could you send the following information;
> 
> * Version of python:   python 3.7.4
> * Is conda or  virtual env or other:   virtual env 
> * logs from the Jupyter output panel
> * Screenshot of the errors

**Comment by zesluo** [comment]
> If I'm remembering right, I believe the problem here was that there are two installs of jupyter and we're only finding the kernelspecs for one and not the other?
> 
> @zesluo can you describe the repro steps? Do we just have to install the Synapse extension?

This problem occurred before(long time ago) and then fixed. now it happens again. I remembered that we were talking about this: " I believe the problem here was that there are two installs of jupyter and we're only finding the kernelspecs for one and not the other", but not sure if it is the final root cause.


repo steps:
1.
![image](https://user-images.githubusercontent.com/46309533/122711920-9bc0dc80-d295-11eb-8d17-a9c6c5e71541.png)
2.
![image](https://user-images.githubusercontent.com/46309533/122711936-a8453500-d295-11eb-96f1-249b2686b75d.png)
this is to trigger this:
![image](https://user-images.githubusercontent.com/46309533/122711965-b5622400-d295-11eb-8560-98a01cdc6443.png)


we just need to install "azure account" and "spark and hive"
![image](https://user-images.githubusercontent.com/46309533/122711822-6b793e00-d295-11eb-9006-f1a84be71713.png)

**Comment by DonJayamanne** [comment]
Please could you upload the contents of the Jupyter output panel.

**Comment by zesluo** [comment]
here it is. and feel free to let me know if a call is needed this week. Thanks!
[jupyteroutputpanel.docx](https://github.com/microsoft/vscode-jupyter/files/6692403/jupyteroutputpanel.docx)

**Comment by DonJayamanne** [comment]
@zesluo Thanks, FYI - in the future uploading a simple text file would be easier than a word document.
Will look into this and if required will ping you.
Thanks

**Comment by zesluo** [comment]
today's log:

User belongs to experiment group 'pythonDiscoveryModuleWithoutWatcher'
User belongs to experiment group 'NativeNotebookEditor'
Info 2021-07-08 08:10:53: ZMQ install verified.
Info 2021-07-08 08:10:53: Attempting to start a server because of preload conditions ...
Info 2021-07-08 08:11:37: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe c:\Users\zesluo.REDMOND\.vscode\extensions\ms-toolsai.jupyter-2021.6.999662501\pythonFiles\normalizeSelection.py
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe c:\Users\zesluo.REDMOND\.vscode\extensions\ms-toolsai.jupyter-2021.6.999662501\pythonFiles\normalizeSelection.py
Info 2021-07-08 08:11:37: Loading webview. View is notset
Info 2021-07-08 08:11:37: Loading web view...
Info 2021-07-08 08:11:37: Webview panel created.
Info 2021-07-08 08:11:37: Waiting for jupyter server and web panel ...
Info 2021-07-08 08:11:37: Creating raw notebook for history://40621ba8-df8d-41ac-bc73-68e8f407ff85
Info 2021-07-08 08:11:37: Getting preferred kernel for history://40621ba8-df8d-41ac-bc73-68e8f407ff85
Info 2021-07-08 08:11:37: Cached data exists getEnvironmentVariables, <No Resource>
Info 2021-07-08 08:11:37: Search all interpreters C:\ZeshiLuo\python\python.exe, C:\Users\zesluo.REDMOND\AppData\Local\Programs\Python\Python38\python.exe, C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:37: initialize CommonMessageCoordinator
Info 2021-07-08 08:11:37: Loading kernelspec from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\share\jupyter\kernels\pysparkkernel\kernel.json for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:37: Loading kernelspec from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\share\jupyter\kernels\python3\kernel.json for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:37: Loading kernelspec from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\share\jupyter\kernels\synapse_pyspark\kernel.json for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:37: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on interpreter path.
Info 2021-07-08 08:11:37: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on path in argv.
Info 2021-07-08 08:11:37: findKernel found Python 3.6.8 64-bit ('hdinsightJupyter': venv)
Info 2021-07-08 08:11:37: Find kernel spec, Class name = T, completed in 182ms, has a truthy return value, Arg 1: <Uri:c:\ZeshiLuo\test\test.py>, Arg 2: undefined, Arg 3: undefined, Return Value: {"kind":"startUsingPythonInterpreter","kernelSpec":{"specFile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\share\\jupyter\\kernels\\python3\\kernel.json","interpreterPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","name":"python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca","argv":["python","-m","ipykernel_launcher","-f","{connection_file}"],"language":"python","path":"python","display_name":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"interpreter":{"sysPrefix":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","envType":"Venv","envName":"hdinsightJupyter","envPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","path":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","architecture":3,"sysVersion":"3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.6.8","major":3,"minor":6,"patch":8,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"id":".jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe"}
Info 2021-07-08 08:11:37: Computing working directory history://40621ba8-df8d-41ac-bc73-68e8f407ff85
Info 2021-07-08 08:11:37: Connecting to raw session for history://40621ba8-df8d-41ac-bc73-68e8f407ff85 with connection {"kind":"startUsingPythonInterpreter","kernelSpec":{"specFile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\share\\jupyter\\kernels\\python3\\kernel.json","interpreterPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","name":"python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca","argv":["python","-m","ipykernel_launcher","-f","{connection_file}"],"language":"python","path":"python","display_name":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"interpreter":{"sysPrefix":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","envType":"Venv","envName":"hdinsightJupyter","envPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","path":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","architecture":3,"sysVersion":"3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.6.8","major":3,"minor":6,"patch":8,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"id":".jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe"}
Info 2021-07-08 08:11:37: Starting raw kernel Python 3.6.8 64-bit ('hdinsightJupyter': venv)
Info 2021-07-08 08:11:37: installMissingDependencies C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:37: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
Info 2021-07-08 08:11:37: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m pip list
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m pip list
Info 2021-07-08 08:11:37: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
Info 2021-07-08 08:11:39: Web view react rendered
Info 2021-07-08 08:11:39: Request for onigasm file at c:\Users\zesluo.REDMOND\.vscode\extensions\ms-toolsai.jupyter-2021.6.999662501\out\client\node_modules\onigasm\lib\onigasm.wasm
Info 2021-07-08 08:11:39: Request for tmlanguage file.
Info 2021-07-08 08:11:39: Searching for token colors ...
Info 2021-07-08 08:11:39: Attempting search for colors ...
Info 2021-07-08 08:11:39: Searching for token colors ...
Info 2021-07-08 08:11:39: Attempting search for colors ...
Info 2021-07-08 08:11:39: Searching for token colors ...
Info 2021-07-08 08:11:39: Attempting search for colors ...
Info 2021-07-08 08:11:39: Searching for token colors ...
Info 2021-07-08 08:11:39: Attempting search for colors ...
Info 2021-07-08 08:11:40: Install Missing Dependencies, Class name = w, completed in 2577ms, has a falsy return value, Return Value: undefined
Info 2021-07-08 08:11:40: Kernel launching with ports 9000,9001,9002,9003,9004. Start port is 9000
Info 2021-07-08 08:11:40: Launching kernel daemon for Python 3.6.8 64-bit ('hdinsightJupyter': venv) # C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:40: No custom variables for Kernel as interpreter is not conda, but is Venv
Info 2021-07-08 08:11:41: Creating daemon process for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe with env variables count 64
Info 2021-07-08 08:11:41: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
Info 2021-07-08 08:11:41: Creating daemon process for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe with env variables count 64
Info 2021-07-08 08:11:41: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Execute rpc method prewarm_kernel in DS Daemon
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Execute rpc method exec_module in DS Daemon
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Execute rpc method prewarm_kernel from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Execute rpc method exec_module from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Pre-Warm DS Kernel in DS Kernel Launcher Daemon
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Exec module in DS Kernel Launcher Daemon ipykernel_launcher with args ['--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"802c9c55-ea97-41a6-a783-24a380ff36cc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-18624eeEZPwky15gX.json']
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Exec in DS Kernel Launcher Daemon (observable) ['C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe', 'c:\\Users\\zesluo.REDMOND\\.vscode\\extensions\\ms-toolsai.jupyter-2021.6.999662501\\pythonFiles\\vscode_datascience_helpers\\kernel_prewarm_starter.py']
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Exec in DS Kernel Launcher Daemon (observable) ipykernel_launcher with args ['--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"802c9c55-ea97-41a6-a783-24a380ff36cc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-18624eeEZPwky15gX.json']
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Exec in DS Kernel Launcher Daemon (observable)
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Exec in DS Kernel Launcher Daemon (observable) ['C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe', '-m', 'ipykernel_launcher', '--ip=127.0.0.1', '--stdin=9003', '--control=9001', '--hb=9000', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"802c9c55-ea97-41a6-a783-24a380ff36cc"', '--shell=9002', '--transport="tcp"', '--iopub=9004', '--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-18624eeEZPwky15gX.json']
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Kernel launched, with PID 15232
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Exec in DS Kernel Launcher Daemon (observable)
Warn 2021-07-08 08:11:41: Python Daemon (pid: 8732): Waiting for Kernel to die 15232
Info 2021-07-08 08:11:41: Python Daemon (pid: 16452): Kernel launched, with PID 19264
Info 2021-07-08 08:11:41: Python Daemon (pid: 8732): Kernel launched, with PID as a daemon 15232
Warn 2021-07-08 08:11:41: Python Daemon (pid: 16452): Waiting for Kernel to die 19264
Info 2021-07-08 08:11:41: Launching kernel daemon, Class name = f, completed in 1869ms, has a truthy return value, Arg 1: <Uri:c:\ZeshiLuo\test\test.py>, Arg 2: "c:\\ZeshiLuo\\test", Arg 3: {"specFile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\share\\jupyter\\kernels\\python3\\kernel.json","interpreterPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","name":"python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca","argv":["python","-m","ipykernel_launcher","--ip=127.0.0.1","--stdin=9003","--control=9001","--hb=9000","--Session.signature_scheme=\"hmac-sha256\"","--Session.key=b\"802c9c55-ea97-41a6-a783-24a380ff36cc\"","--shell=9002","--transport=\"tcp\"","--iopub=9004","--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-18624eeEZPwky15gX.json"],"language":"python","path":"python","display_name":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"}, Arg 4: {"sysPrefix":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","envType":"Venv","envName":"hdinsightJupyter","envPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","path":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","architecture":3,"sysVersion":"3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.6.8","major":3,"minor":6,"patch":8,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"}, Return Value: <Return value cannot be serialized for logging>
Info 2021-07-08 08:11:41: Launching kernel in kernelProcess.ts, Class name = b, completed in 1870ms, has a truthy return value, Arg 1: "c:\\ZeshiLuo\\test", Return Value: {"proc":{"_events":{},"_eventsCount":2,"_closesNeeded":3,"_closesGot":0,"connected":false,"signalCode":null,"exitCode":null,"killed":false,"spawnfile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","_handle":{"pid":16452},"spawnargs":["C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","-m","vscode_datascience_helpers.daemon","--daemon-module=vscode_datascience_helpers.kernel_launcher_daemon","-v"],"pid":16452,"stdin":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":null,"ended":false,"endEmitted":false,"reading":false,"sync":true,"needReadable":false,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null,"readable":false},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":false,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stdout":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{"close":[null,null]},"_eventsCount":4,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stderr":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stdio":[{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":null,"ended":false,"endEmitted":false,"reading":false,"sync":true,"needReadable":false,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null,"readable":false},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":false,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{"close":[null,null]},"_eventsCount":4,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null}]},"out":{"_isScalar":false,"observers":[],"closed":false,"isStopped":false,"hasError":false,"thrownError":null}}
Info 2021-07-08 08:11:42: Searching for token colors ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Searching for token colors ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Searching for token colors ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Searching for token colors ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Attempting search for colors ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Loading base colors from c:\Program Files\Microsoft VS Code\resources\app\extensions\theme-defaults\themes\dark_plus.json ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Using colors to generate CSS ...
Info 2021-07-08 08:11:42: Attempting to start a server because of preload conditions ...
Info 2021-07-08 08:11:44: Raw session started and connected
Started kernel Python 3.6.8 64-bit ('hdinsightJupyter': venv)
Info 2021-07-08 08:11:44: Initial setup for history://40621ba8-df8d-41ac-bc73-68e8f407ff85 starting ...
Info 2021-07-08 08:11:44: UpdateWorkingDirectoryAndPath
Info 2021-07-08 08:11:44: Kernel switching to busy
Info 2021-07-08 08:11:44: Kernel switching to idle
Info 2021-07-08 08:11:44: Starting raw kernel Python 3.6.8 64-bit ('hdinsightJupyter': venv)
Info 2021-07-08 08:11:44: installMissingDependencies C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:44: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
Info 2021-07-08 08:11:44: Kernel switching to busy
Info 2021-07-08 08:11:44: Kernel switching to idle
Info 2021-07-08 08:11:44: Initial setup complete for history://40621ba8-df8d-41ac-bc73-68e8f407ff85
Info 2021-07-08 08:11:44: Finished connecting a2f90353-a74e-4c9b-a18e-08782acbb945
Info 2021-07-08 08:11:44: IPyWidgetScriptSource.initialize
Info 2021-07-08 08:11:44: Adding sys info for eb9bf610-cbce-4107-a759-2efeaaa8c6a8 0
Info 2021-07-08 08:11:44:  [32mQueuing messages (no listenerts)
Info 2021-07-08 08:11:44: Registering commtarget jupyter.widget
Info 2021-07-08 08:11:44: IPyWidgetMessageDispatcher.initialize
Info 2021-07-08 08:11:44: Sys info for eb9bf610-cbce-4107-a759-2efeaaa8c6a8 0 complete
Info 2021-07-08 08:11:44: Created and initailized CommonMessageCoordinator
Info 2021-07-08 08:11:44: Submitting code for eb9bf610-cbce-4107-a759-2efeaaa8c6a8
Info 2021-07-08 08:11:44: Wait for sys info for eb9bf610-cbce-4107-a759-2efeaaa8c6a8 0
Info 2021-07-08 08:11:44: UpdateWorkingDirectoryAndPath
Info 2021-07-08 08:11:44: changeDirectoryIfPossible
Info 2021-07-08 08:11:44: Kernel switching to busy
Info 2021-07-08 08:11:44: Kernel switching to idle
Info 2021-07-08 08:11:44: IPyWidgetMessageDispatcher.initialize
Info 2021-07-08 08:11:44: IPyWidgetMessageDispatcher.initialize
Info 2021-07-08 08:11:44: Kernel switching to busy
Info 2021-07-08 08:11:44: Kernel switching to idle
Info 2021-07-08 08:11:44: Adding hash for 1 = 9edcb9b1cffd with 20 lines
Info 2021-07-08 08:11:44: Kernel switching to busy
Info 2021-07-08 08:11:44: Finished execution for 778ad216-70a0-4ecd-90be-0d6edd3348d7
Info 2021-07-08 08:11:44: Kernel switching to idle
Info 2021-07-08 08:11:46: Install Missing Dependencies, Class name = w, completed in 1473ms, has a falsy return value, Return Value: undefined
Info 2021-07-08 08:11:46: Kernel launching with ports 9005,9006,9007,9008,9009. Start port is 9000
Info 2021-07-08 08:11:46: Launching kernel daemon for Python 3.6.8 64-bit ('hdinsightJupyter': venv) # C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:46: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import ipykernel"
Info 2021-07-08 08:11:46: No custom variables for Kernel as interpreter is not conda, but is Venv
Info 2021-07-08 08:11:46: Python Daemon (pid: 8732): Execute rpc method start_prewarmed_kernel in DS Daemon
Info 2021-07-08 08:11:46: Python Daemon (pid: 8732): Execute rpc method start_prewarmed_kernel from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:46: Python Daemon (pid: 8732): Start pre-warmed Kernel in DS Kernel Launcher Daemon 15232 with args ['ipykernel_launcher', '--ip=127.0.0.1', '--stdin=9008', '--control=9006', '--hb=9005', '--Session.signature_scheme="hmac-sha256"', '--Session.key=b"c1950aff-b87f-4ff4-984a-7bbb9efeb036"', '--shell=9007', '--transport="tcp"', '--iopub=9009', '--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-1862418HE62E2HMGq.json']
Info 2021-07-08 08:11:46: Launching kernel daemon, Class name = f, completed in 32ms, has a truthy return value, Arg 1: <Uri:c:\ZeshiLuo\test\test.py>, Arg 2: "c:\\ZeshiLuo\\test", Arg 3: {"specFile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\share\\jupyter\\kernels\\python3\\kernel.json","interpreterPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","name":"python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca","argv":["python","-m","ipykernel_launcher","--ip=127.0.0.1","--stdin=9008","--control=9006","--hb=9005","--Session.signature_scheme=\"hmac-sha256\"","--Session.key=b\"c1950aff-b87f-4ff4-984a-7bbb9efeb036\"","--shell=9007","--transport=\"tcp\"","--iopub=9009","--f=C:\\Users\\ZESLUO~1.RED\\AppData\\Local\\Temp\\tmp-1862418HE62E2HMGq.json"],"language":"python","path":"python","display_name":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"}, Arg 4: {"sysPrefix":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","envType":"Venv","envName":"hdinsightJupyter","envPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","path":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","architecture":3,"sysVersion":"3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.6.8","major":3,"minor":6,"patch":8,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"}, Return Value: <Return value cannot be serialized for logging>
Info 2021-07-08 08:11:46: Launching kernel in kernelProcess.ts, Class name = b, completed in 33ms, has a truthy return value, Arg 1: "c:\\ZeshiLuo\\test", Return Value: {"proc":{"_events":{},"_eventsCount":2,"_closesNeeded":3,"_closesGot":0,"connected":false,"signalCode":null,"exitCode":null,"killed":false,"spawnfile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","_handle":{"pid":8732},"spawnargs":["C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","-m","vscode_datascience_helpers.daemon","--daemon-module=vscode_datascience_helpers.kernel_launcher_daemon","-v"],"pid":8732,"stdin":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":null,"ended":false,"endEmitted":false,"reading":false,"sync":true,"needReadable":false,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null,"readable":false},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":false,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stdout":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{"close":[null,null]},"_eventsCount":4,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stderr":{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},"stdio":[{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":null,"ended":false,"endEmitted":false,"reading":false,"sync":true,"needReadable":false,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null,"readable":false},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":false,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{"close":[null,null]},"_eventsCount":4,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null},{"connecting":false,"_hadError":false,"_parent":null,"_host":null,"_readableState":{"objectMode":false,"highWaterMark":16384,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":true,"ended":false,"endEmitted":false,"reading":true,"sync":false,"needReadable":true,"emittedReadable":false,"readableListening":false,"resumeScheduled":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"destroyed":false,"errored":null,"closed":false,"closeEmitted":false,"defaultEncoding":"utf8","awaitDrainWriters":null,"multiAwaitDrain":false,"readingMore":false,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3,"_writableState":{"objectMode":false,"highWaterMark":16384,"finalCalled":false,"needDrain":false,"ending":false,"ended":false,"finished":false,"destroyed":false,"decodeStrings":false,"defaultEncoding":"utf8","length":0,"writing":false,"corked":0,"sync":true,"bufferProcessing":false,"writecb":null,"writelen":0,"afterWriteTickInfo":null,"buffered":[],"bufferedIndex":0,"allBuffers":true,"allNoop":true,"pendingcb":0,"prefinished":false,"errorEmitted":false,"emitClose":false,"autoDestroy":false,"errored":null,"closed":false,"writable":false,"closeEmitted":false},"allowHalfOpen":false,"_sockname":null,"_pendingData":null,"_pendingEncoding":"","server":null,"_server":null}]},"out":{"_isScalar":false,"observers":[],"closed":false,"isStopped":false,"hasError":false,"thrownError":null}}
Info 2021-07-08 08:11:47: Creating daemon process for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe with env variables count 64
Info 2021-07-08 08:11:47: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Execute rpc method prewarm_kernel in DS Daemon
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Execute rpc method prewarm_kernel from C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Pre-Warm DS Kernel in DS Kernel Launcher Daemon
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Exec in DS Kernel Launcher Daemon (observable) ['C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe', 'c:\\Users\\zesluo.REDMOND\\.vscode\\extensions\\ms-toolsai.jupyter-2021.6.999662501\\pythonFiles\\vscode_datascience_helpers\\kernel_prewarm_starter.py']
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Exec in DS Kernel Launcher Daemon (observable)
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Kernel launched, with PID 6480
Warn 2021-07-08 08:11:47: Python Daemon (pid: 4136): Waiting for Kernel to die 6480
Info 2021-07-08 08:11:47: Python Daemon (pid: 4136): Kernel launched, with PID as a daemon 6480
Info 2021-07-08 08:12:18: Kernel switching to busy
Info 2021-07-08 08:12:18: Kernel switching to idle
Info 2021-07-08 08:12:29: Cached data exists getEnvironmentVariables, <No Resource>
Info 2021-07-08 08:12:29: notebook communications already initialized for editor untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: Attempting to start a server because of preload conditions ...
Info 2021-07-08 08:12:29: Search all interpreters C:\ZeshiLuo\python\python.exe, C:\Users\zesluo.REDMOND\AppData\Local\Programs\Python\Python38\python.exe, C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:12:29: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on interpreter path.
Info 2021-07-08 08:12:29: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on path in argv.
Info 2021-07-08 08:12:29: findPreferredKernel score for Python3, Python 3.6.8 64-bit ('hdinsightJupyter': venv) is 14
Info 2021-07-08 08:12:29: findPreferredKernel score for Python 3.6.8 64-bit ('hdinsightJupyter': venv) is 24
Info 2021-07-08 08:12:29: findPreferredKernel score for Python3, Python 3.6.8 64-bit is 4
Info 2021-07-08 08:12:29: findPreferredKernel score for Python 3.6.8 64-bit is 4
Info 2021-07-08 08:12:29: findPreferredKernel score for Python3, Python 3.8.8 64-bit is 4
Info 2021-07-08 08:12:29: findPreferredKernel score for Python 3.8.8 64-bit is 4
Info 2021-07-08 08:12:29: findKernel found Python 3.6.8 64-bit ('hdinsightJupyter': venv)
Info 2021-07-08 08:12:29: Find kernel spec, Class name = T, completed in 67ms, has a truthy return value, Arg 1: <Uri:Untitled-1.ipynb>, Arg 2: {"language_info":{"name":"python","nbconvert_exporter":"python"},"orig_nbformat":4}, Arg 3: {"_isCancelled":false,"_emitter":null}, Return Value: {"kind":"startUsingPythonInterpreter","kernelSpec":{"specFile":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\share\\jupyter\\kernels\\pysparkkernel\\kernel.json","interpreterPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","name":"python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca","argv":["python","-m","sparkmagic.kernels.pysparkkernel.pysparkkernel","-f","{connection_file}"],"language":"python","path":"python","display_name":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"interpreter":{"sysPrefix":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","envType":"Venv","envName":"hdinsightJupyter","envPath":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter","path":"C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe","architecture":3,"sysVersion":"3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]","version":{"raw":"3.6.8","major":3,"minor":6,"patch":8,"build":[],"prerelease":["final","0"]},"displayName":"Python 3.6.8 64-bit ('hdinsightJupyter': venv)"},"id":".jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe.C:\\Users\\zesluo.REDMOND\\.msvscode.hdinsight\\hdinsightJupyter\\Scripts\\python.exe"}
Info 2021-07-08 08:12:29: PreferredConnection: .jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe found for NotebookDocument: untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: Cached data exists getEnvironmentVariables, <No Resource>
Info 2021-07-08 08:12:29: Search all interpreters C:\ZeshiLuo\python\python.exe, C:\Users\zesluo.REDMOND\AppData\Local\Programs\Python\Python38\python.exe, C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:12:29: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on interpreter path.
Info 2021-07-08 08:12:29: Kernel python368jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca matches Python 3.6.8 64-bit ('hdinsightJupyter': venv) based on path in argv.
Info 2021-07-08 08:12:29: TargetController found ID: .jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe for document untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: Setting controller affinity for untitled:Untitled-1.ipynb .jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe
Info 2021-07-08 08:12:29: Setting setActiveController for untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: Intiailize notebook communications for editor untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: Resolving notebook UI Comms (resolve) for untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29: initialize CommonMessageCoordinator
Info 2021-07-08 08:12:29: KernelProvider switched kernel to id = .jvsc74a57bd0dee351fbdea7624e10915cb4f30ea41d8713e4035c417d26ccd3173e014f7bca.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe.C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe}
Info 2021-07-08 08:12:29: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m pip list
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m pip list
Info 2021-07-08 08:12:29: IPyWidgetMessageDispatcher.initialize
Info 2021-07-08 08:12:29: Created and initailized CommonMessageCoordinator
Info 2021-07-08 08:12:29: Attach Coordinator for untitled:Untitled-1.ipynb
Info 2021-07-08 08:12:29:  [32mPosting message to Notebook UI 
Info 2021-07-08 08:12:30: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import jupyter"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import jupyter"
Info 2021-07-08 08:12:30: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import notebook"
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -c "import notebook"
Info 2021-07-08 08:12:30: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m jupyter kernelspec --version
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m jupyter kernelspec --version
Info 2021-07-08 08:12:31: Creating daemon pool for C:\Users\zesluo.REDMOND\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe with env variables count 64
Info 2021-07-08 08:12:31: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v
Info 2021-07-08 08:12:31: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v
Info 2021-07-08 08:12:31: Process Execution: > ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v
> ~\.msvscode.hdinsight\hdinsightJupyter\Scripts\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.jupyter_daemon -v

**Comment by zesluo** [comment]
![image](https://user-images.githubusercontent.com/46309533/124849806-08481500-dfd2-11eb-87dd-dd9c1f4a20d9.png)

**Comment by DavidKutu** [comment]
@zesluo, sorry for the late reply. @DonJayamanne merged the fix on July 7th, and you seem to have tested it that same day. 
His fix was on our main branch on July 7th but would've hit VS Code Insiders until July 8th at around 2am PST.

This should work on both stable and insiders. Could you please test if works for you?

**Comment by DonJayamanne** [comment]
I'm closing this issue as it was fixed, the Python extension is not listing the Python environment, thats an issue with the Python extension.
I have already communicated this to @zesluo via email as well.
